test/phoenix_live_view_test.exs

```diff
@@ -342,6 +342,22 @@ defmodule Phoenix.LiveViewUnitTest do
     end
   end
 
+  describe "update with arity 2 function" do
+    test "passes socket assigns to update function" do
+      socket = @socket |> assign(key: "value", key2: "another") |> Utils.clear_changed()
+
+      socket = update(socket, :key2, fn _, %{key: key} -> key end)
```

Comment:
It might be worth adding a test using both args to cover all cases.

Reply:
Makes sense. Thank you